cmake/modules/CephChecks.cmake

```diff
@@ -92,14 +92,23 @@ CHECK_STRUCT_HAS_MEMBER("struct stat" st_mtim.tv_nsec sys/stat.h
 CHECK_STRUCT_HAS_MEMBER("struct stat" st_mtimespec.tv_nsec sys/stat.h
   HAVE_STAT_ST_MTIMESPEC_TV_NSEC LANGUAGE C)
 
-if(CMAKE_SYSTEM_PROCESSOR STREQUAL CMAKE_HOST_SYSTEM_PROCESSOR)
+if(NOT CMAKE_CROSSCOMPILING AND CMAKE_SYSTEM_PROCESSOR STREQUAL CMAKE_HOST_SYSTEM_PROCESSOR)
```

Comment:
after reading https://cmake.org/cmake/help/v3.1/manual/cmake-toolchains.7.html#cross-compiling-for-linux and https://github.com/Kitware/CMake/blob/master/Modules/CMakeDetermineSystem.cmake, i think `CMAKE_CROSSCOMPILING` is better than `CMAKE_SYSTEM_PROCESSOR STREQUAL CMAKE_HOST_SYSTEM_PROCESSOR`, so could you move this change into a separated commit to s/CMAKE_SYSTEM_PROCESSOR STREQUAL CMAKE_HOST_SYSTEM_PROCESSOR/NOT CMAKE_CROSSCOMPILING/ ?

Reply:
From what I can tell, CMAKE_CROSSCOMPILING wouldn't be set if the target cpu architecture doesn't match the host one. One idea would be to set it ourselves, what do you think?

e.g.

```
if(NOT(CMAKE_SYSTEM_PROCESSOR STREQUAL CMAKE_HOST_SYSTEM_PROCESSOR))
   set(CMAKE_CROSSCOMPILING TRUE)
endif()
```

Reply:
agreed, but if a developer tries to crosscompile a piece of software using cmake, he/she is very likely to use a toolchain file which set `CMAKE_SYSTEM_NAME` manually. so can we assume that we can always rely on this assumption?

see https://cmake.org/cmake/help/latest/variable/CMAKE_CROSSCOMPILING.html . IIUC this variable is set by cmake automatically. and normally,

Reply:
Ok, I think that's reasonable.